op-signer/gen-local-creds.sh

```diff
@@ -0,0 +1,189 @@
+        -sha256 \
+        -out "$CA_CERT" \
+        -keyout "$CA_KEY" \
+        -subj "/O=$CERT_ORG_NAME/CN=root"
+}
+
+generate_client_tls() {
+    local hostname="$1"
+    echo
+    echo "Generating client TLS credentials for $hostname..."
+    
+    
+    # Create a directory for this client's credentials
+    local clientDir="$TLS_DIR/$hostname"
+    mkdir -p "$clientDir"
+    
+    # Generate client key
+    echo "Generating client key..."
+    run_openssl genrsa -out "$clientDir/$CLIENT_TLS_KEY" "$MOD_LENGTH"
+
+    local confFile="$clientDir/$CLIENT_OPENSSL_CNF"
+    
+    # Create a config file for the CSR
+    cat > "$confFile" << EOF
+[req]
+distinguished_name=req
+[san]
+subjectAltName=DNS:$hostname
+EOF
+    
+    echo "Generating client certificate signing request..."
+    run_openssl req -new -key "$clientDir/$CLIENT_TLS_KEY" \
+        -sha256 \
+        -out "$clientDir/$CLIENT_TLS_CSR" \
+        -subj "/O=$CERT_ORG_NAME/CN=$hostname" \
+        -extensions san \
+        -config "$confFile"
+    
+    echo "Generating client certificate..."
+    run_openssl x509 -req -in "$clientDir/$CLIENT_TLS_CSR" \
+        -sha256 \
+        -CA "$CA_CERT" \
+        -CAkey "$CA_KEY" \
+        -CAcreateserial \
+        -out "$clientDir/$CLIENT_TLS_CERT" \
+        -days 3 \
+        -extensions san \
+        -extfile "$confFile"
+}
+
+generate_client_key() {
+    local hostname="$1"
+    echo
+    echo "Generating private key for $hostname..."
+    local clientDir="$TLS_DIR/$hostname"
+    mkdir -p "$clientDir"
+    run_openssl ecparam -name secp256k1 -genkey -noout -param_enc explicit \
+        -out "$clientDir/$CLIENT_PRIVATE_KEY"
+}
+
+
+generate_client_credentials() {
+    setup_client_hostnames "$@"
+    process_clients generate_client_tls
+    process_clients generate_client_key
+}
+
+setup_client_hostnames() {
+    CLIENT_HOSTNAMES="$*"
+    if [ -z "$CLIENT_HOSTNAMES" ]; then
+        CLIENT_HOSTNAMES="localhost"
+    fi
+    printf "\nProcessing clients: %s\n" "$CLIENT_HOSTNAMES"
+}
+
+process_clients() {
+    generator="$1"
+    for hostname in $CLIENT_HOSTNAMES; do
+        "$generator" "$hostname"
+    done
+}
+
+# Valid targets for the script
+VALID_TARGETS="ca, client, client_tls, client_key, all"
+
+# Get target and client hostnames from command line arguments
+if [ $# -eq 0 ]; then
+    echo "Error: Target argument is required. Must be one of: $VALID_TARGETS"
+    exit 1
+fi
+
+TARGET="$1"; shift
+
+echo "----------------------------------------"
+echo "!!!! DO NOT USE IN PRODUCTION !!!!!"
+echo "This script is meant for development/testing ONLY."
+echo "Generating credentials..."
+echo
+echo "Target: $TARGET"
+echo "Using Docker: $USE_DOCKER"
+echo "----------------------------------------"
+
+mkdir -p "$TLS_DIR"
+
+case "$TARGET" in
+    "ca")
+        generate_ca true
+        ;;
+    "client_tls")
+        setup_client_hostnames "$@"
+        process_clients generate_client_tls
+        ;;
+    "client_key")
```

Comment:
Same here. How about 

```suggestion
    "signing_key")
```